prevent openpilot from engaging with no lead at zero speed on toyotas

cydia2020 opened the pull request:
**Description**
fixes #19801

cydia2020 committed:
prevent openpilot from engaging with no lead at zero speed on toyotas

pd0wm commented on selfdrive/car/toyota/carcontroller.py:
```diff
@@ -107,7 +107,7 @@ def update(self, enabled, CS, frame, actuators, pcm_cancel_cmd, hud_alert,
     # we can spam can to cancel the system even if we are using lat only control
     if (frame % 3 == 0 and CS.CP.openpilotLongitudinalControl) or (pcm_cancel_cmd and Ecu.fwdCamera in self.fake_ecus):
-      lead = lead or CS.out.vEgo < 12.    # at low speed we always assume the lead is present do ACC can be engaged
+      lead = lead or (CS.out.vEgo < 12. and CS.out.vEgo > 0.001)    # at low speed we always assume the lead is present so ACC can be engaged
```
```suggestion
      lead = lead or (CS.out.vEgo < 12. and (not CS.out.standstill))    # at low speed we always assume the lead is present so ACC can be engaged
```

pd0wm replied:
The standstill field is probably better suited. Does that also work correctly?

cydia2020 replied:
One sec, let me test this

cydia2020 replied:
This doesn't work, op can still engage at zero speed

pd0wm replied:
Strange. The threshold for standstill is the same (0.001). Only thing is it uses raw wheel speed instead of the Kalman filered one.

Can you make it work with the vEgoRaw, but with a larger threshold? The Kalman filter can behave a little erratic at very low speed causing vEgo to not be correct.

cydia2020 replied:
It might be because openpilot mistaking my bushes as a lead, which happened before. I can try again on road this morning

cydia2020 replied:
OK this does work.

cydia2020 committed:
use standstill instead of vego > 0.001

cydia2020 committed:
always assume lead is present when enabled

regardless of current speed, this is to prevent op from accidently disengaging at low speed when enabled if it lose track of the lead vehicle for some reason. (tested and working)
